azure-quantum/azure/quantum/workspace.py

```diff
@@ -223,13 +224,28 @@ def get_job(self, job_id: str) -> Job:
         details = client.get(job_id)
         return Job(self, details)
 
-    def list_jobs(self) -> List[Job]:
+    def list_jobs(
+        self, 
+        name_match: str = None, 
+        status: Optional[JobStatus] = None,
+        created_after: Optional[datetime] = None
+    ) -> List[Job]:
+        """Returns list of jobs that meet optional (limited) filter criteria. 
+            :param name_match: regex expression for job name matching
+            :param status: filter by job status
+            :param created_after: filter jobs after time of job creation
+
+            todo #27666 - replace some of these with server-side filtering.  
+            todo #27664 - tag filtering when swagger is ready

```

Comment:
I would recommend not putting todo's in the code but instead track them in ADO and link this file there